src/api/flow.ts

```diff
@@ -173,11 +115,13 @@ export function createFlowGenerator(name: string, generator: Function) {
     return function() {
         const ctx = this
         const args = arguments
-        return new Promise(function(resolve, reject) {
-            const runId = ++generatorId
+        const runId = ++generatorId
+        const gen = action(`${name} - runid: ${runId} - init`, generator).apply(ctx, args)
+        let rejector: (error: any) => void
+
+        const res = new Promise(function(resolve, reject) {
```

Comment:
I think we should check if `gen` returned a promise (which is the case if it's an async generator) and then `then` it with `next` rather than passing `gen` to next.

That way, you can use async generators and yield for state changes and avoid accidental changes.

I still think it would be really awesome if you could `await` for async stuff but `yield` for state updates of an observable.

Reply:
> I still think it would be really awesome if you could await for async stuff but yield for state updates of an observable. 

Me2, but different story I think :)

Reply:
That would just require checking if `gen` returns a promise when you call `.next` (which would make it an async iterator) and then wait for it - that's all it takes given the implementation since `await`s would work anyway and `yield` would run in an action because of the code above. Cancellation would also work for sub actions just the same.

Reply:
Added support for async iterators